### PR TITLE
Fixing dom structure of Tags to follow

### PR DESCRIPTION
**Pull request by Rafi993**
## What type of PR is this?

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Suggested tag during onboarding screen is made screen reader friendly with this PR. more context in https://github.com/forem/forem/pull/12384#discussion_r576524771

## Related Tickets & Documents

closes https://github.com/forem/forem/issues/9585

## QA Instructions, Screenshots, Recordings

1. Go to `/onboarding`
2. Go to screen "What are you interested in?"
3. Enable voice over and use keyboard to navigate the app

### UI accessibility concerns?

This PR will make tags to follow during onboarding more accessible

## Added tests?

- [ ] Yes
- [x] No, and this is why: It is a html change
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [X] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

### CODE REVIEW

**Comment by aitchiss** [comment]
Hey @Rafi993 - thanks for this PR and apologies for the delay in review! I took some time to explore whether or not we are definitely taking the right approach here with the checkboxes, as it felt a little off for keyboard navigation that we now have to activate by `space` and not `Enter` (as a sighted user would expect from looking at the UI).

After a bit of research and a chat with @nickytonline I think actually we need to reverse the approach and use `button` elements with the `aria-pressed` attribute to convey the state. This will make sure that:
- The keyboard interaction behaviour matches up with what is presented visually
- Screen reader users will get the additional context on whether the item is currently selected or not

I appreciate this was the approach you initially took, and we asked you to implement with the checkboxes instead - I do apologise for that - we did think it was best at the time and didn't mean to send you down the wrong path! We really appreciate the time put into all your contributions, and thank you for bearing with us on this one. We're always learning! 

Let me know if you're still OK to work on this and change back to buttons with the additional `aria`. Thanks again for your patience ❤️

**Comment by Rafi993** [comment]
> Hey @Rafi993 - thanks for this PR and apologies for the delay in review! I took some time to explore whether or not we are definitely taking the right approach here with the checkboxes, as it felt a little off for keyboard navigation that we now have to activate by `space` and not `Enter` (as a sighted user would expect from looking at the UI).
> 
> After a bit of research and a chat with @nickytonline I think actually we need to reverse the approach and use `button` elements with the `aria-pressed` attribute to convey the state. This will make sure that:
> 
> * The keyboard interaction behaviour matches up with what is presented visually
> * Screen reader users will get the additional context on whether the item is currently selected or not
> 
> I appreciate this was the approach you initially took, and we asked you to implement with the checkboxes instead - I do apologise for that - we did think it was best at the time and didn't mean to send you down the wrong path! We really appreciate the time put into all your contributions, and thank you for bearing with us on this one. We're always learning!
> 
> Let me know if you're still OK to work on this and change back to buttons with the additional `aria`. Thanks again for your patience ❤️

Let me look at this later this week. Thanks for your review. I would love to look at this.

**Comment by citizen428** [comment]
Hi @Rafi993, thanks for your contribution. No rush, but I wanted to check in and see how things are going with this PR on your end?

**Comment by Rafi993** [comment]
> Hi @Rafi993, thanks for your contribution. No rush, but I wanted to check in and see how things are going with this PR on your end?

Hi sorry for the delay. I have made the suggested changes now.

**Comment by citizen428** [comment]
No need at all to be sorry, @Rafi993, thanks for making the changes 💯

**Review by citizen428 — APPROVED** [review]

File: app/javascript/onboarding/components/FollowTags.jsx [hunk 1]
```diff
@@ -130,53 +130,45 @@ export class FollowTags extends Component {
...
-                          : tag.bg_color_hex,
-                        color: selectedTags.includes(tag)
-                          ? tag.bg_color_hex
-                          : tag.text_color_hex,
-                      }}
-                    >
-                      {selectedTags.includes(tag) ? (
-                        <span>
-                          <span className="onboarding-tags__button-default">
-                            ✓ Following
```
Comment (aitchiss) [inline app/javascript/onboarding/components/FollowTags.jsx:167]: nitpick (blocking) - just one small thing - the checkmark that previously showed when you are following a tag is no longer appearing. Previously it was added like this, but I think it'd be an improvement to use the checkmark svg in combination with the `crayons-btn--icon-left` class ([like this example in storybook](https://storybook.forem.com/?path=/story/components-buttons-html--icon-left))

**Review by aitchiss — CHANGES_REQUESTED** [review]
This is looking great @Rafi993! I tested locally with Safari/VoiceOver and it's a big improvement - nice one 😄 

I just have one small change request - it'll be great to see this merged soon 🚀

File: app/javascript/onboarding/components/FollowTags.jsx (hunk 1 above)
Comment (Rafi993) [reply app/javascript/onboarding/components/FollowTags.jsx:167]: Thanks for you time @aitchiss I have made the changes

Screenshot 2021-05-03 at 7 02 21 PM

File: app/javascript/onboarding/components/FollowTags.jsx [hunk 2]
```diff
@@ -130,53 +130,53 @@ export class FollowTags extends Component {
...
+                        }}
+                      >
+                        <svg
+                          width="24"
+                          height="24"
+                          xmlns="http://www.w3.org/2000/svg"
+                          className="crayons-icon"
+                        >
+                          <path d="M9.99999 15.172L19.192 5.979L20.607 7.393L9.99999 18L3.63599 11.636L5.04999 10.222L9.99999 15.172Z" />
+                        </svg>
```
Comment (aitchiss) [inline app/javascript/onboarding/components/FollowTags.jsx:173]: At the moment, the icon is showing even if you're not following - I haven't tested this, but I _think_ it should do the trick

```suggestion
                        { selected && (<svg
                          width="24"
                          height="24"
                          xmlns="http://www.w3.org/2000/svg"
                          className="crayons-icon"
                        >
                          <path d="M9.99999 15.172L19.192 5.979L20.607 7.393L9.99999 18L3.63599 11.636L5.04999 10.222L9.99999 15.172Z" />
                        </svg>)}
```

File: app/javascript/onboarding/components/FollowTags.jsx [hunk 3]
```diff
@@ -130,53 +130,53 @@ export class FollowTags extends Component {
...
+                    }}
+                  >
+                    <div className="onboarding-tags__item__inner">
+                      #{tag.name}
+                      <button
+                        type="button"
+                        onClick={() => this.handleClick(tag)}
+                        className={`onboarding-tags__button crayons-btn--icon-left ${
+                          selected && 'onboarding-tags__button--selected'
+                        }`}
```
Comment (aitchiss) [inline app/javascript/onboarding/components/FollowTags.jsx:154]: See below - I think we only want this class when the button is selected

```suggestion
                        className={`onboarding-tags__button  ${
                          selected && 'onboarding-tags__button--selected crayons-btn--icon-left'
                        }`}
```

**Review by aitchiss — CHANGES_REQUESTED** [review]
Nearly there! just a minor thing to tweak 😄

**Comment by Rafi993** [comment]
> Nearly there! just a minor thing to tweak 😄

Sorry I missed it before.

File: app/javascript/onboarding/components/FollowTags.jsx (hunk 3 above)
Comment (Rafi993) [reply app/javascript/onboarding/components/FollowTags.jsx]: This is how it looks after the changes

Tags

**Review by msarit — APPROVED** [review]
Tested locally and all works as expected. Also ensured that all requested changes have been implemented. 👍🏾  🚀

**Review by aitchiss — APPROVED** [review]
Looks awesome! Thanks for this @Rafi993, and for sticking with the back and forth on the approach ❤️

**Comment by Rafi993** [comment]
> Looks awesome! Thanks for this @Rafi993, and for sticking with the back and forth on the approach ❤️

Thank you @aitchiss for your review

**Pull request merged**